### PR TITLE
Change in data model in test_theseus_layer

### PR DESCRIPTION
**Pull request by dishank-b**
## Motivation and Context

The data model was wrong as in it had to two optima, leading to some of the optimizer not able to pass the test. The comments in the file also suggests that the intentional model and actually implemented model were different.
## How Has This Been Tested

All the current optimizer passes the test with the changed model. 

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dishank-b** [comment]
@mhmukadam @luisenp, made changes in the `test_theseus_layer`. They were causing bug for DCEM convergence. 

Also, in the  comments in the file, it say the model we want to have is `f(x) = Ax^2+b` whereas the one implemented was `f(x) = (Ax)^2+b`. I corrected it now. Though verify the change in the grad once. I feel the grad in previous version also wrong for the model we had.

File: tests/test_theseus_layer.py [hunk 1]
```diff
@@ -26,12 +26,12 @@
...
 def model(x, b):
-    return (b[..., :1] * x) ** 2 + b[..., 1:]
+    return b[..., :1] * x**2 + b[..., 1:]
 
 
 def model_grad(x, b):
-    g1 = 2 * (b[..., :1] * x) * x
-    g2 = b[..., 1:] * torch.ones_like(x)
+    g1 = x**2
+    g2 = torch.ones_like(x)
```
Comment (mhmukadam) [inline tests/test_theseus_layer.py:34]: Should the size here be related to `b[..., 1:]`?

**Review by mhmukadam — APPROVED** [review]
LGTM, thanks for the fix. Just one comment.

File: tests/test_theseus_layer.py (hunk 1 above)
Comment (luisenp) [reply tests/test_theseus_layer.py:34]: Thanks for catching this, not sure what I was thinking when I added that `b` factor. The size is correct, because of the following:

- `b` (optim_var) is shape `(batch_size, 2)`
- `x` (data, constant) is shape `(batch_size, num_points)`
- error is shape `(batch_size, num_points)`

We want the jacobian with respect to `b` be of shape `(batch_size, num_points, 2)`, and it's computed below as the stack of `[g1, g2]` on `axis=2`, so we need for `g2` to be of shape `(batch_size, num_points)`, which is the shape of `x`.

**Review by luisenp — APPROVED** [review]
Thanks for the fix!

**Comment by dishank-b** [comment]
Thanks. Can we merge this then?  Then I will rebase the DCEM PR on top of this.
